crates/ruff_linter/src/rule_selector.rs

```diff
@@ -289,6 +289,8 @@ mod schema {
                                 (!prefix.is_empty()).then(|| prefix.to_string())
                             })),
                     )
+                    // Strip the test rules from the schema
+                    .filter(|code| !code.starts_with("RUF9"))
```

Comment:
Wish there were something better here (e.g., what if we could use the rule source, and have a special `lint_source()` called `test`), but might be difficult.

Reply:
Should we just put them all in a separate lint group e.g. `RFT` for "Ruff test"? Seems safer and easier to filter in general?

Reply:
Might end up being more work to implement overall though?

Reply:
I'm actually wondering how these even up getting included in the schema, do we run with `--all-features` or something?

Reply:
Anyway, I'm cool with whatever here.

Reply:
Huh this raises a preview warning

```
❯ ./target/debug/ruff check example.py --select RUF9
warning: Selection `RUF9` has no effect because the `--preview` flag was not included.
```

but doesn't do anything else.. `RUF8` errors

```
❯ ./target/debug/ruff check example.py --select RUF8
error: invalid value 'RUF8' for '--select <RULE_CODE>'
```

I guess I'll fix that next..

Reply:
Yeah this is actually confusing... why are they included in the schema in the first place?

Reply:
They shouldn't even _exist_ when we run `cargo dev generate-all`, right?

Reply:
Okay this was a bug in the proc macro 🎉 
https://github.com/astral-sh/ruff/pull/9747/commits/8247b8582b4b80900a0ef7e984a035283b4558b6